src/gtc/dace/nodes.py

```diff
@@ -134,7 +135,8 @@ def __init__(
                 for j, he in enumerate(section.horizontal_executions):
                     extents_dict[j * len(oir_node.sections) + i] = extents[id(he)]
 
-            self.oir_node = oir_node
+            # TODO: Why is this conversion required?
```

Comment:
Why is there a TODO here? Just looking at the type hints, it seems that the `oir_node` argument is a `VerticalLoop` instance but the `self.oir_node` is a `PickledDataclassProperty`, so clearly a conversion is required, right?

I guess the `@library.node` decorator is creating instance attributes in a similar way to `dataclasses`, because otherwise all the `self.oir_node` (and other attributes) are just assigning to the class instance `type(self).oir_node`

Reply:
I removed the todo. Thanks for catching! That was just one to myself as I was editing.